vcf-dumper/vcf-dumper-ws/src/main/java/uk/ac/ebi/eva/vcfdump/server/HtsgetVcfController.java

```diff
@@ -0,0 +1,213 @@
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MultivaluedHashMap;
+import javax.ws.rs.core.MultivaluedMap;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * Created by amilamanoj on 13.05.17.
+ */
+@RestController
+@RequestMapping(value = "/variants/")
+@Api(tags = {"htsget"})
+public class HtsgetVcfController {
+
+    private static final String VCF = "VCF";
+
+    private Properties evaProperties;
+
+    public HtsgetVcfController() throws IOException {
+        evaProperties = new Properties();
+        evaProperties.load(VcfDumperWSServer.class.getResourceAsStream("/eva.properties"));
+    }
+
+    @RequestMapping(value = "/{id}/", method = RequestMethod.GET, produces = "application/json; charset=UTF-8")
+    public ResponseEntity getHtsgetUrls(
+            @PathVariable("id") String id,
+            @RequestParam(name = "format", required = false) String format,
+            @RequestParam(name = "referenceName", required = false) String referenceName,
+            @RequestParam(name = "species", required = false) String species,
+            @RequestParam(name = "start", required = false) Integer start,
+            @RequestParam(name = "end", required = false) Integer end,
+            @RequestParam(name = "fields", required = false) List<String> fields,
+            @RequestParam(name = "tags", required = false, defaultValue = "") String tags,
+            @RequestParam(name = "notags", required = false, defaultValue = "") String notags,
+            HttpServletRequest request,
+            HttpServletResponse response)
+            throws IllegalAccessException, IllegalOpenCGACredentialsException, InstantiationException, IOException,
+            StorageManagerException, URISyntaxException, ClassNotFoundException {
+
+        if (!VCF.equals(format)) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .body(new HtsGetError("UnsupportedFormat", "Specified format is not supported by this server"));
+        }
+
+        if (start != null && end != null && end <= start) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .body(new HtsGetError("InvalidRange", "The requested range cannot be satisfied"));
+        }
+
+        String dbName = "eva_" + species;
+        String regionStr = referenceName + ((start != null && end != null) ? ":" + start + "-" + end : "");
+        MultivaluedMap<String, String> queryParameters = new MultivaluedHashMap<>();
+        queryParameters.put(VariantDBAdaptor.REGION, Collections
+                .singletonList(regionStr));
+
+        HtsGetResponse htsGetResponse = new HtsGetResponse();
```

Comment:
this line and the following one can be moved further down, so all the htsGetResponse operations are together. It could be even refactored and extracted all to a private method buildHtsGetReponse(request, id, referenceName, ...)